edgraph/access_ee_test.go

```diff
@@ -0,0 +1,180 @@
+			"namespace": namespace,
+			"userid":    userId,
+			"exp":       e1,
+		})
+	}
+
+	tokenString, err := token.SignedString([]byte(worker.Config.HmacSecret))
+	if err != nil {
+		_ = fmt.Errorf("Something Went Wrong: %s", err.Error())
+		return "", err
+	}
+	return tokenString, nil
+}
+
+func sliceCompare(x, y []string) bool {
+	if len(x) != len(y) {
+		return false
+	}
+
+	for i, s := range x {
+		if s != y[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+func TestValidateToken(t *testing.T) {
+	var (
+		err         error
+		tokenString string
+	)
+
+	for _, userdata := range userDataList {
+		if tokenString, err = generateJWT(userdata.namespace, userdata.userId, userdata.groupIds, 0); err != nil {
+			t.Errorf("JWT token string generation error: %s", err)
+			return
+		}
+
+		ud, err := validateToken(tokenString)
+		if err != nil {
+			t.Errorf("validateToken() error: %s", err)
+			return
+		}
+		if ud.namespace != userdata.namespace || ud.userId != userdata.userId || !sliceCompare(ud.groupIds, userdata.groupIds) {
+			t.Errorf("Actual output %+v is not equal to the expected output %+v", userdata, ud)
+		}
+	}
+}
+
+func TestGetAccessJwt(t *testing.T) {
+	var (
+		err    error
+		tokstr string
+		jwtstr string
+		exp    int64 = time.Now().Add(worker.Config.AccessJwtTtl).Unix()
+	)
+
+	grpLst := []acl.Group{
+		{
+			Uid:     "100",
+			GroupID: "1001",
+			Users:   []acl.User{},
+			Rules:   []acl.Acl{},
+		},
+		{
+			Uid:     "101",
+			GroupID: "1011",
+			Users:   []acl.User{},
+			Rules:   []acl.Acl{},
+		},
+		{
+			Uid:     "102",
+			GroupID: "1021",
+			Users:   []acl.User{},
+			Rules:   []acl.Acl{},
+		},
+	}
+
+	for _, userdata := range userDataList {
+		g := acl.GetGroupIDs(grpLst)
+		if tokstr, err = generateJWT(userdata.namespace, userdata.userId, g, exp); err != nil {
+			t.Errorf("JWT token string generation error: %s", err)
+			return
+		}
+		t1 := strings.Split(tokstr, ".")
+
+		jwtstr, err = getAccessJwt(userdata.userId, grpLst, userdata.namespace)
+		if err != nil {
+			t.Errorf("getAccessJwt () error: %s", err)
+			return
+		}
+		t2 := strings.Split(jwtstr, ".")
+
+		/*
+		 * JWT token format = [header].[payload].[signature].
+		 * One of the factor [signature] would differ is based on the expiry time.
+		 * This will differ for tokstr and jwtstr and hence ignoring it.
+		 */
+		if t1[0] != t2[0] || t1[1] != t2[1] {
+			t.Errorf("Actual output is not equal to the expected output")
+		}
+	}
+}
+
+func TestGetRefreshJwt(t *testing.T) {
+	var (
+		err    error
+		tokstr string
+		jwtstr string
+		exp    int64 = time.Now().Add(worker.Config.RefreshJwtTtl).Unix()
+	)
+
+	for _, userdata := range userDataList {
+		if tokstr, err = generateJWT(userdata.namespace, userdata.userId, nil, exp); err != nil {
+			t.Errorf("JWT token string generation error: %s", err)
+			return
+		}
+		t1 := strings.Split(tokstr, ".")
+
+		jwtstr, err = getRefreshJwt(userdata.userId, userdata.namespace)
+		if err != nil {
```

Comment:
We don't need to do error handling in tests, because if it errors out, the test will fail, which is good enough.